examples/hybrid/advection_agnesi.jl

```diff
@@ -0,0 +1,376 @@
+    warp_fn = warp_agnesi,
+);
+# pressure
+function pressure(ρθ)
+    if ρθ >= 0
+        return MSLP * (R_d * ρθ / MSLP)^γ
+    else
+        return NaN
+    end
+end
+
+# geopotential
+function Φ(z)
+    grav * z
+end
+
+function rayleigh_sponge(
+    z;
+    z_sponge = 12000.0,
+    z_max = 15000.0,
+    α = 0.5,  # Relaxation timescale
+    τ = 0.5,
+    γ = 2.0,
+)
+    if z >= z_sponge
+        r = (z - z_sponge) / (z_max - z_sponge)
+        β_sponge = α * sinpi(τ * r)^γ
+        return β_sponge
+    else
+        return eltype(z)(0)
+    end
+end
+
+# Reference: https://journals.ametsoc.org/view/journals/mwre/140/4/mwr-d-10-05073.1.xml, Section
+function init_agnesi_2d(x, z)
+    θ₀ = 280.0
+    cp_d = C_p
+    cv_d = C_v
+    p₀ = MSLP
+    g = grav
+    γ = cp_d / cv_d
+
+    𝒩 = 0.01
+    π_exner = @. exp(-g * z / (cp_d * θ₀))
+    θ = @. θ₀ * exp(𝒩^2 * z / g)
+    ρ = @. p₀ / (R_d * θ) * (π_exner)^(cp_d / R_d)
+    ρθ = @. ρ * θ
+    ρuₕ = @. ρ * Geometry.UVector(10.0)
+
+    return (ρ = ρ, ρθ = ρθ, ρuₕ = ρuₕ)
+end
+
+# initial conditions
+coords = Fields.coordinate_field(hv_center_space)
+face_coords = Fields.coordinate_field(hv_face_space)
+
+Yc = map(coords) do coord
+    init_agnesi_2d(coord.x, coord.z)
+end
+
+ρw = map(face_coords) do coord
+    Geometry.WVector(0.0)
+end
+
+Y = Fields.FieldVector(Yc = Yc, ρw = ρw)
+
+function energy(Yc, ρu, z)
+    ρ = Yc.ρ
+    ρθ = Yc.ρθ
+    u = ρu / ρ
+    kinetic = ρ * norm(u)^2 / 2
+    potential = z * grav * ρ
+    internal = C_v * pressure(ρθ) / R_d
+    return kinetic + potential + internal
+end
+
+function combine_momentum(ρuₕ, ρw)
+    Geometry.transform(Geometry.UWAxis(), ρuₕ) +
+    Geometry.transform(Geometry.UWAxis(), ρw)
+end
+
+function center_momentum(Y)
+    If2c = Operators.InterpolateF2C()
+    combine_momentum.(Y.Yc.ρuₕ, If2c.(Y.ρw))
+end
+
+function total_energy(Y)
+    ρ = Y.Yc.ρ
+    ρu = center_momentum(Y)
+    z = Fields.coordinate_field(axes(ρ)).z
+    sum(energy.(Yc, ρu, z))
+end
+
+function rhs!(dY, Y, params, t)
+    ρw = Y.ρw
+    Yc = Y.Yc
+    dYc = dY.Yc
+    dρw = dY.ρw
+
+    # spectral horizontal operators
+    hdiv = Operators.Divergence()
+    hgrad = Operators.Gradient()
+
+    # vertical FD operators with BC's
+    vvdivc2f = Operators.DivergenceC2F(
+        bottom = Operators.SetDivergence(Geometry.WVector(0.0)),
+        top = Operators.SetDivergence(Geometry.WVector(0.0)),
+    )
+    uvdivf2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(
+            Geometry.WVector(0.0) ⊗ Geometry.UVector(0.0),
+        ),
+        top = Operators.SetValue(Geometry.WVector(0.0) ⊗ Geometry.UVector(0.0)),
+    )
+    If = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    Ic = Operators.InterpolateF2C()
+    ∂ = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(0.0)),
+        top = Operators.SetValue(Geometry.WVector(0.0)),
```

Comment:
This is true for the flat domain, but I don't think this is right for the warped mesh case - since `u \dot n = 0` is what we wish to enforce such that the flux normal to the surface is zero. @Zhengyu-Huang